src/hotspot/share/jeandle/jeandleAbstractInterpreter.cpp

```diff
@@ -1410,6 +1410,23 @@ llvm::Value* JeandleAbstractInterpreter::find_or_insert_oop(ciObject* oop) {
   return oop_value;
 }
 
+llvm::Value* JeandleAbstractInterpreter::resolve_klass(ciKlass* ciklass, bool will_link, int klass_index) {
+  // Check if the class is loaded and will link at compile time
+  if (will_link && ciklass->is_loaded()) {
+    // Convert ciKlass to runtime Klass pointer
+    Klass* klass = (Klass*)(ciklass->constant_encoding());
+
+    llvm::PointerType* klass_type = llvm::PointerType::get(*_context, llvm::jeandle::AddrSpace::CHeapAddrSpace);
+    llvm::Value* klass_addr = _ir_builder.getInt64((intptr_t)klass);
+    llvm::Value* klass_ptr = _ir_builder.CreateIntToPtr(klass_addr, klass_type);
+    return klass_ptr;
+  } else {
+    // Class has not been resolved, call runtime class resolution
+    llvm::Value* index_value = _ir_builder.getInt32(klass_index);
+    return call_java_op("jeandle.resolve_klass", {index_value});
```

Comment:
According to C2's implementation, I think it's better to insert an uncommon_trap here, instead of resolve klass.